operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -0,0 +1,722 @@
+    self.writeln("    listenPort: " + str(nap.getListenPort()))
+    self.writeln("    publicPort: " + str(nap.getPublicPort()))
+
+
+class BootPropertiesGenerator(Generator):
+
+  def __init__(self, env):
+    Generator.__init__(self, env, env.BOOT_FILE)
+
+  def generate(self):
+    self.open()
+    try:
+      self.addBootProperties()
+      self.close()
+      self.addGeneratedFile()
+    finally:
+      self.close()
+
+  def addBootProperties(self):
+    self.writeln("username=" + self.readAndEncryptSecret("username"))
+    self.writeln("password=" + self.readAndEncryptSecret("password"))
+
+class UserConfigAndKeyGenerator(Generator):
+
+  def __init__(self, env):
+    Generator.__init__(self, env, env.USERKEY_FILE)
+
+  def generate(self):
+    self.open()
+    try:
+      # first, generate UserConfig file and add it, also generate UserKey file
+      self.addUserConfigAndKey()
+      self.close()
+      # now add UserKey file
+      self.addGeneratedFile()
+    finally:
+      self.close()
+
+  def addUserConfigAndKey(self):
+    username = self.readSecret("username")
+    password = self.readSecret("password")
+    nm_host = 'localhost'
+    nm_port = '5556' 
+    domain_name = self.env.getDomain().getName()
+    domain_home = self.env.getDomainHome()
+    isNodeManager = "true"
+    userConfigFile = self.env.USERCONFIG_FILE
+    userKeyFileBin = self.env.USERKEY_FILE + '.bin'
+
+    trace("nmConnect " + username + ", " + nm_host + ", " + nm_port + ", " + domain_name + ", " + domain_home)
+    nmConnect(username, password, nm_host, nm_port, domain_name, domain_home, 'plain')
+    try:
+      # the following storeUserConfig WLST command generates two files:
+      storeUserConfig(userConfigFile, userKeyFileBin, isNodeManager)
+
+      # user config is already a text file, so directly add it to the generated file list
+      self.env.addGeneratedFile(userConfigFile)
+
+      # but key is a binary, so we b64 it - and the caller of this method will add the file
+      userKey = self.env.readBinaryFile(userKeyFileBin)
+      b64 = ""
+      for s in base64.encodestring(userKey).splitlines():
+        b64 = b64 + s
+      self.writeln(b64)
+    finally:
+      nmDisconnect()
+
+class SitConfigGenerator(Generator):
+
+  def __init__(self, env):
+    Generator.__init__(self, env, env.CM_FILE)
+
+  def generate(self):
+    self.open()
+    try:
+      self.addSitCfg()
+      self.close()
+      self.addGeneratedFile()
+    finally:
+      self.close()
+
+  def addSitCfg(self):
+    self.addSitCfgXml()
+
+  def addSitCfgXml(self):
+    self.writeln("<?xml version='1.0' encoding='UTF-8'?>")
+    self.writeln("<d:domain xmlns:d=\"http://xmlns.oracle.com/weblogic/domain\" xmlns:f=\"http://xmlns.oracle.com/weblogic/domain-fragment\" xmlns:s=\"http://xmlns.oracle.com/weblogic/situational-config\">")
+    self.indent()
+    self.writeln("<s:expiration> 2999-07-16T19:20+01:00 </s:expiration>")
```

Comment:
In src191000, you can just skip the expiration element.  Not sure about earlier branches.

Reply:
Yes, let's test without this.  We're going to support 12.2.1.3, but only with patches for sit config.

Reply:
Good points.  Thanks.  I've made the change and tests still pass.